plugins/modules/consul_role.py

```diff
@@ -0,0 +1,636 @@
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    node_identities:
+      - name: node-1
+        datacenter: dc2
+
+- name: Remove a role
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    state: absent
+"""
+
+RETURN = """
+role:
+    description: The role object.
+    returned: success
+    type: dict
+    sample:
+        {
+            "CreateIndex": 39,
+            "Description": "",
+            "Hash": "Trt0QJtxVEfvTTIcdTUbIJRr6Dsi6E4EcwSFxx9tCYM=",
+            "ID": "9a300b8d-48db-b720-8544-a37c0f5dafb5",
+            "ModifyIndex": 39,
+            "Name": "foo-role",
+            "Policies": [
+                {"ID": "b1a00172-d7a1-0e66-a12e-7a4045c4b774", "Name": "foo-access"}
+            ]
+        }
+operation:
+    description: The operation performed on the role.
+    returned: changed
+    type: str
+    sample: update
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import missing_required_lib
+from ansible_collections.community.general.plugins.module_utils.consul import (
+    get_consul_url, get_auth_headers, handle_consul_response_error)
+import traceback
+
+REQUESTS_IMP_ERR = None
+
+try:
+    from requests.exceptions import ConnectionError
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+    REQUESTS_IMP_ERR = traceback.format_exc()
+
+TOKEN_PARAMETER_NAME = "token"
+HOST_PARAMETER_NAME = "host"
+SCHEME_PARAMETER_NAME = "scheme"
+VALIDATE_CERTS_PARAMETER_NAME = "validate_certs"
+NAME_PARAMETER_NAME = "name"
+DESCRIPTION_PARAMETER_NAME = "description"
+PORT_PARAMETER_NAME = "port"
+POLICIES_PARAMETER_NAME = "policies"
+SERVICE_IDENTITIES_PARAMETER_NAME = "service_identities"
+NODE_IDENTITIES_PARAMETER_NAME = "node_identities"
+STATE_PARAMETER_NAME = "state"
+
+PRESENT_STATE_VALUE = "present"
+ABSENT_STATE_VALUE = "absent"
+
+REMOVE_OPERATION = "remove"
+UPDATE_OPERATION = "update"
+CREATE_OPERATION = "create"
+
+POLICY_RULE_SPEC = dict(
+    name=dict(type='str'),
+    id=dict(type='str'),
+)
+
+NODE_ID_RULE_SPEC = dict(
+    name=dict(type='str', required=True),
+    datacenter=dict(type='str', required=True),
+)
+
+SERVICE_ID_RULE_SPEC = dict(
+    name=dict(type='str', required=True),
+    datacenters=dict(type='list', elements='str', required=True),
+)
+
+_ARGUMENT_SPEC = {
+    TOKEN_PARAMETER_NAME: dict(no_log=True),
+    PORT_PARAMETER_NAME: dict(default=8500, type='int'),
+    HOST_PARAMETER_NAME: dict(default='localhost'),
+    SCHEME_PARAMETER_NAME: dict(default='http'),
+    VALIDATE_CERTS_PARAMETER_NAME: dict(type='bool', default=True),
+    NAME_PARAMETER_NAME: dict(required=True),
+    DESCRIPTION_PARAMETER_NAME: dict(required=False, type='str', default=''),
+    POLICIES_PARAMETER_NAME: dict(type='list', elements='dict', options=POLICY_RULE_SPEC, default=None),
```

Comment:
Please don't forget
```suggestion
    POLICIES_PARAMETER_NAME: dict(type='list', elements='dict', options=POLICY_RULE_SPEC, mutually_exclusive=[('name', 'id')], required_one_of=[('name', 'id')], default=None),
```
to make sure that `name` or `id` (but not both) have to be specified.

(See https://github.com/ansible-collections/community.general/pull/6972#discussion_r1306355348)

Reply:
Thanks, I oversaw that suggestion previously